packages/ui/src/contexts/SwapProvider/useEvmSwap.tsx

```diff
@@ -484,6 +526,77 @@ export const useEvmSwap: SwapAdapter<OptimalRate> = (
     ],
   );
 
+  const performWrapOperation = useCallback(
+    async (params: SwapParams<WrapOperation> | SwapParams<UnwrapOperation>) => {
+      assertPresent(network, CommonError.NoActiveNetwork);
+      assertPresent(account, CommonError.NoActiveAccount);
+      assertPresent(rpcProvider, CommonError.Unknown);
+      assert(!network.isTestnet, CommonError.UnknownNetwork);
+
+      const { srcToken, destToken, srcDecimals, destDecimals } = params;
+      const userAddress = account.addressC;
+      const tokenAddress =
+        params.quote.type === 'WRAP'
```

Comment:
Please consider to use the guard type function you created 😄 
```suggestion
        isWrapOperationParams(params)
```

Reply:
23aa31cd6